main/screen.c

```diff
@@ -295,6 +320,14 @@ static void screen_update_cb(lv_timer_t * timer)
             display_on(true);
         }
     }
+    
+    // Update WiFi RSSI periodically
+    if (GLOBAL_STATE->SYSTEM_MODULE.is_connected) {
+        int8_t rssi_value;
+        if (get_wifi_current_rssi(&rssi_value) == ESP_OK) {
+            GLOBAL_STATE->SYSTEM_MODULE.wifi_rssi = rssi_value;
```

Comment:
This is in the same function as where it's used. I think the `wifi_rssi` field can be removed from SYSTEM_MODULE, and the call can be done further down.